ipa-core/src/query/runner/hybrid.rs

```diff
@@ -20,31 +20,40 @@ use crate::{
         step::ProtocolStep::Hybrid,
     },
     query::runner::reshard_tag::reshard_aad,
-    report::hybrid::{
-        EncryptedHybridReport, IndistinguishableHybridReport, UniqueTag, UniqueTagValidator,
+    report::{
+        hybrid::{
+            EncryptedHybridReport, IndistinguishableHybridReport, UniqueTag, UniqueTagValidator,
+        },
+        hybrid_info::HybridInfo,
     },
     secret_sharing::replicated::semi_honest::AdditiveShare as Replicated,
 };
 
 #[allow(dead_code)]
-pub struct Query<C, HV, R: PrivateKeyRegistry> {
+pub struct Query<'a, C, HV, R: PrivateKeyRegistry> {
     config: HybridQueryParams,
     key_registry: Arc<R>,
+    hybrid_info: HybridInfo<'a>,
```

Comment:
I probably missed that on the previous review, but helper origin is assumed to be known [statically](https://github.com/private-attribution/ipa/blob/9477db8a38f4bf40cfb946ab59960cee04a9f69c/ipa-core/src/report/hybrid.rs#L60) to helpers. This should help to avoid the lifetime proliferation and the need to keep it inside query struct

If there is a need for having it, we can assume helper origin is `&'static str`

Reply:
The design doc for this project has the helper origin as part of both HybridConversionInfo and HybridImpressionInfo, so I assume it's needed? I can try static though

Reply:
yes, it is definitely needed, but it is assumed to be per MPC ring, i.e. known in advance. We hardcoded it in our Rust implementation, but generally I think `&static str` would be more appropriate

Reply:
Is conversion_site_domain also static? If so, we can remove the lifetime

Reply:
Conversion site is dynamic and provided by the submitter when they want to execute this query